src/Kestrel.Core/Internal/Http/HttpUpgradeStream.cs

```diff
@@ -145,6 +145,13 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
             return _requestStream.ReadAsync(buffer, offset, count, cancellationToken);
         }
 
+#if NETCOREAPP2_1
+        public override ValueTask<int> ReadAsync(Memory<byte> destination, CancellationToken cancellationToken = default)
+        {
+            return _requestStream.ReadAsync(destination, cancellationToken);
+        }
+#endif
```

Comment:
List all of the current TFMs so we know if this gets outdated. E.g. this breaks if we add 2.2.

Reply:
I'm not sure exactly what you're asking for here. But if you want to change this, we should probably change the pattern [everywhere else](https://github.com/aspnet/KestrelHttpServer/search?q=%23if+NETCOREAPP2_1&unscoped_q=%23if+NETCOREAPP2_1) too.

Reply:
This pattern:
https://github.com/aspnet/MusicStore/blob/d971bfc8b777ca4098a951727218aef32cb9ebd5/test/MusicStore.E2ETests/Common/Helpers.cs#L51-L57
This is used pretty broadly in other repos to avoid dead code and regressions when TFMs change.

Reply:
I see. Do you think we should change the pattern everywhere or just one place since it fails compilation?

Reply:
Everywhere, but it could be a seperate PR.